Sources/C++Workshop/Cards.h

```diff
@@ -137,28 +137,28 @@ class Player {
     using HandIndex = typename Hand::size_type;
     
     /// Remove and return a card from the player's hand at the given index.
-    auto play(HandIndex index) -> Card<Deck>& {
-        auto& card = this->_hand.at(index).get();
+    auto play(HandIndex index) -> std::unique_ptr<Card<Deck>> {
+        auto card = std::move(this->_hand.at(index));
```

Comment:
Q: what's the benefit of adding `std::move()` here? (I've checked [its cppreference page](https://en.cppreference.com/w/cpp/utility/move) but still not totally understand it's purpose yet)

Reply:
I would love to get some explanation around `unique_ptr` and `move` around here

Reply:
The notes on the `move` page, plus the [`unique_ptr` constructor](https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr) #5 explain this kind of obliquely, but basically:

Only one `std::unique_ptr<Card>` is allowed to point to a given object, so `unique_ptr`s aren't allowed to be copied. Instead, they need to be _moved_ (i.e. the new one is constructed using that #5 _move constructor_, which the spec says nulls out the original pointer). In order to trigger the move constructor, we need to pass an _rvalue_ (in this case, `std::unique_ptr<Card>&&`, note the double-ampersand), which we get by calling `std::move(...)`; it effectively casts the _lvalue_ (regular variable or normal reference to one) to an rvalue.